script.js

```diff
@@ -1,57 +1,83 @@
-const add = document.getElementById('add');
-const awesomeBooks = [];
+const StoredBooks = [];
 
-add.addEventListener('click', (e) => {
-  e.preventDefault();
-  const title = document.getElementById('title');
-  const author = document.getElementById('author');
+class Display {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 
-  const book = {
-    Title: title.value,
-    Author: author.value,
-  };
-  const list = document.getElementById('book-list');
-  const newDiv = document.createElement('div');
-  newDiv.classList.add('book');
+  static addtolist() {
+    const Title = document.getElementById('title').value;
+    const Author = document.getElementById('author').value;
+    const list = document.getElementById('book-list');
+    const newDiv = document.createElement('div');
+    newDiv.classList.add('book');
+    const newbook = new Display(Title, Author);
+    StoredBooks.push(newbook);
+
+    newDiv.innerHTML += `
+    <span>${newbook.title}</span><span> by</span><span> ${newbook.author}</span>
+      <button class="delete">Delete</button>
+    `;
+
+    list.appendChild(newDiv);
+    document.getElementById('title').value = '';
+    document.getElementById('author').value = '';
+  }
+
+  static addtolocal() {
+    localStorage.setItem('books', JSON.stringify(StoredBooks));
+  }
+
+  static delBook(e) {
+    e.preventDefault();
+    if (e.target.classList.contains('delete')) {
+      e.target.parentElement.remove();
+    }
+  }
 
-  awesomeBooks.push(book);
+  static removelocal(e) {
+    StoredBooks.forEach((newbook, i) => {
+      if (
+        e.target.parentElement.firstElementChild.textContent === newbook.title
+      ) {
+        StoredBooks.splice(i, 1);
+      }
```

Comment:
- [ ] (Optional) It would be nice to use the `filter` method instead of the `splice` method to remove a book. That way, you do not mutate the original array. It is also considered good practice to not mutate arrays when changing states. 👍